activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -540,9 +540,18 @@ def disable_referential_integrity
       end
 
       # Override to check all foreign key constraints in a database.
+      # DEPRECATED: Override
+      # <tt>ActiveRecord::ConnectionAdapters#check_all_foreign_keys_valid</tt>
+      # to raise a useful error message instead
       def all_foreign_keys_valid?
         true
       end
+      deprecate :all_foreign_keys_valid?
+
+      # Override to check all foreign key constraints in a database.
+      def check_all_foreign_keys_valid
```

Comment:
I don't see a way of doing this without breaking 3rd party adapters. An alternative would be to create a custom deprecator that we can "turn off" during the `check_all_foreign_keys_valid` method?

Reply:
I've removed the dependency. This will mean that fixture integrity checks in 3rd party adapters that implement `all_foreign_keys_valid?` will no longer work until they override `check_all_foreign_keys_valid` instead.

Reply:
I'm not sure if I understand your concern yet, How about creating another private method `_all_foreign_keys_valid?`?

```ruby
def all_foreign_keys_valid?
    # Just call the private `_all_foreign_keys_valid` method
    _all_foreign_keys_valid?
end
deprecate :all_foreign_keys_valid?

def check_all_foreign_keys_valid
    # Making use of the private `_all_foreign_keys_valid?` method
    # so that it does not depend on the deprecated public `all_foreign_keys_valid?` method
end

private 
def _all_foreign_keys_valid?
    # implement the current `all_foreign_keys_valid?` method logic here
end
```
Then we can remove the public `all_foreign_keys_valid?` method in future.

Reply:
Aha, good idea, thank you 🙏🏻

Reply:
Wait, that makes no difference since there isn't any real "logic" in the abstract adapter, and this doesn't solve the issue with 3rd party abstract adapters, since they will still be overriding `all_foreign_keys_valid?`, which will no longer be used by the fixtures integrity check. Unless we just don't deprecate it until we find a way to remove it safely.

How do you deprecate a method for "overriding"? Unless we do some reflection to see if the current adapter has overridden `all_foreign_keys_valid?` and not `check_all_foreign_keys_valid` and raise a deprecation warning if that's the case. e.g. something like:

```ruby
def all_foreign_keys_valid?
  true
end

def check_all_foreign_keys_valid
  return if self.class.method(:all_foreign_keys_valid?).owner == AbstractAdapter
  
  # log some kind of deprecation warning
  raise if all_foreign_keys_valid?
end
```

Reply:
I am also a 3rd party adapter maintainer and there is no way for every third party adapter to call `super` so we do not have to something like `return if self.class.method(:all_foreign_keys_valid?).owner == AbstractAdapter` .
Here is one example https://github.com/rsim/oracle-enhanced/pull/1770 Also we can notify by adding mentions to some of 3rd party adapter maintainer if necessary.

Reply:
Okay, in that case the current implementation should be sufficient